openwisp_monitoring/device/admin.py

```diff
@@ -93,30 +128,64 @@ def get_queryset(self, request):
         return super().get_queryset(request).order_by('created')
 
     def has_add_permission(self, request, obj=None):
-        return False
+        return super().has_add_permission(request, obj) or request.user.has_perm(
+            f'{self.model._meta.app_label}.add_alertsettings_inline'
+        )
+
+    def has_view_permission(self, request, obj=None):
+        return super().has_view_permission(request, obj) or request.user.has_perm(
+            f'{self.model._meta.app_label}.view_alertsettings_inline'
+        )
+
+    def has_change_permission(self, request, obj=None):
+        return super().has_change_permission(request, obj) or request.user.has_perm(
+            f'{self.model._meta.app_label}.change_alertsettings_inline'
+        )
 
     def has_delete_permission(self, request, obj=None):
-        return False
+        return super().has_delete_permission(request, obj) or request.user.has_perm(
+            f'{self.model._meta.app_label}.delete_alertsettings_inline'
+        )
 
 
 class MetricInline(NestedGenericStackedInline):
     model = Metric
     extra = 0
     inlines = [AlertSettingsInline]
-    readonly_fields = ['name', 'is_healthy']
-    fields = ['name', 'is_healthy']
+    fields = ['name', 'is_healthy', 'field_name', 'configuration']
+    readonly_fields = ['is_healthy']
     # Explicitly changed name from Metrics to Alert Settings
     verbose_name = _('Alert Settings')
     verbose_name_plural = verbose_name
 
+    def get_fields(self, request, obj=None):
+        if not self.has_change_permission(request, obj) or not self.has_view_permission(
+            request, obj
+        ):
+            return ['name', 'is_healthy']
+        return super().get_fields(request, obj)
+
     def has_add_permission(self, request, obj=None):
-        return False
+        # User will be able to add alertsettings from inline
+        # even if it only has permission to add a alertsettings object
+        return super().has_add_permission(request, obj) or request.user.has_perm(
+            f'{self.model._meta.app_label}.add_alertsettings_inline'
+        )
 
-    def has_delete_permission(self, request, obj=None):
-        return False
+    def has_view_permission(self, request, obj=None):
+        return super().has_view_permission(request, obj) or request.user.has_perm(
+            f'{self.model._meta.app_label}.view_alertsettings_inline'
+        )
 
-    def get_queryset(self, request):
-        return super().get_queryset(request).filter(alertsettings__isnull=False)
+    def has_change_permission(self, request, obj=None):
+        return super().has_change_permission(request, obj) or request.user.has_perm(
+            f'{self.model._meta.app_label}.change_alertsettings_inline'
+        )
+
+    def has_delete_permission(self, request, obj=None):
+        return super().has_delete_permission(request, obj) or request.user.has_perm(
+            f'{self.model._meta.app_label}.delete_alertsettings_inline'
+        )
```

Comment:
These methods to check for the additional permissions are always the same, except the suffix of the permission, therefore we could create a base class / mixin which has this logic and requires only a class attribute to be set which specifies the suffix, eg: `inline_permission_suffix = 'alertsettings_inline'`.

This allows us to follow the DRY principle closely and reduce maintenance whenever we need to change or upgrade this logic.